src/wallet/wallet.cpp

```diff
@@ -2220,22 +2221,22 @@ SpendableInputs CWallet::FindSpendableInputs(
     SpendableInputs unspent;
     for (auto const& [wtxid, wtx] : mapWallet) {
         bool isCoinbase = wtx.IsCoinBase();
-        auto nDepth = wtx.GetDepthInMainChain();
+        auto nDepth = wtx.GetDepthInMainChain(asOfHeight);
 
         // Filter the transactions before checking for coins
         if (!CheckFinalTx(wtx)) continue;
         if (nDepth < 0 || nDepth < minDepth) continue;
 
         if (selectTransparent &&
             // skip transparent utxo selection if coinbase spend restrictions are not met
-            (!isCoinbase || (allowTransparentCoinbase && wtx.GetBlocksToMaturity() <= 0))) {
+            (!isCoinbase || (allowTransparentCoinbase && wtx.GetBlocksToMaturity(asOfHeight) <= 0))) {
 
             for (int i = 0; i < wtx.vout.size(); i++) {
                 const auto& output = wtx.vout[i];
                 isminetype mine = IsMine(output);
 
                 // skip spent utxos
-                if (IsSpent(wtxid, i)) continue;
+                if (IsSpent(wtxid, i, std::nullopt)) continue;
```

Comment:
This skips UTXOs that are currently spent, rather than UTXOs which were spent at the given height. It's unclear to me that this will result in the behaviour that is documented for the RPCs that use this.

Reply:
You’re right. This is a mistake. Should use `asOfHeight`.